MonoDevelop.MSBuild.Editor/Completion/MSBuildCompletionCommitManager.cs

```diff
@@ -33,6 +36,34 @@ public bool ShouldCommitCompletion (IAsyncCompletionSession session, SnapshotPoi
 
 		public CommitResult TryCommit (IAsyncCompletionSession session, ITextBuffer buffer, CompletionItem item, char typedChar, CancellationToken token)
 		{
+			string insertionText;
+			var completionKind = item.Properties.PropertyList[item.Properties.PropertyList.Count - 1].Value;
```

Comment:
This is extremely fragile. You should access the property by key, not assume it's the last one.

Reply:
Fixed it. See my latest commit.